Assets/Scripts/R&D Tree/RDTreeManager.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class RDTreeManager : MonoBehaviour
+{
+    [SerializeField] private List<ResearchNode> _firstAvailableNodes = new List<ResearchNode>();
+     
+    public ResearchNode CurrentResearching { get; private set; }
+    public List<ResearchNode> AllNodes { get; set; }
+    public List<ResearchNode> ResearchQueue {  get; private set; }
+
+    private void Awake()
+    {
+        AllNodes = new List<ResearchNode>();
+        ResearchQueue = new List<ResearchNode>();
+    }
+
+    private void Start() => ActivateFirstNodes();    
+
+    private void FixedUpdate() => CurrentResearching?.AddValue(1);   
+
+    /// <summary>
+    /// This function sets the firstavailable nodes in the tree active and available
+    /// </summary>
+    private void ActivateFirstNodes()
+    {
+        foreach(ResearchNode node in _firstAvailableNodes)
+        {
+            node.SetAvailableAvailable();
+        }
+    }
+
+    /// <summary>
+    /// Function that returns a list of all the nodes that are available to buy
+    /// </summary>
+    /// <returns>Return a list with all the available nodes</returns>
+    public List<ResearchNode> AvailableNodes()
+    {
+        var availableNodes = new List<ResearchNode>();
+
+        foreach(var node in AllNodes)
+        {
+            if(node.CurrentResearchState == ResearchNode.ResearchStates.available)
+                availableNodes.Add(node);
+        }
+
+        return availableNodes;
+    }    
+
+    /// <summary>
+    /// Checks if there is an node in the queue and returns that node
+    /// </summary>
+    /// <returns>Returns the next node in the queue</returns>
+    public ResearchNode NextInQueue()
+    {
+        if (ResearchQueue.Count > 0)
+            return ResearchQueue[0];
+        else
+            return null;
+    }    
+    
+    public void StartNewResearch(ResearchNode research)
+    {
+        if(research.CurrentResearchState != ResearchNode.ResearchStates.available)
+            return;
+
+        ChooseNewResearch(research);
+        research.SetResearchInDevelopment();
```

Comment:
I think the `research.SetResearchInDevelopment();` could be integrated into the `ChooseNewResearch(research);` function since it's closely bound